iothub/device/tests/UtilsTests.cs

```diff
@@ -35,5 +36,64 @@ public void ConvertDeliveryAckTypeToStringInvalidValueFail()
         {
             Utils.ConvertDeliveryAckTypeToString((DeliveryAcknowledgement)100500);
         }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void MergeDictionaries_OneItemInListIsNull()
+        {
+            var dict1 = new Dictionary<string, string> { { "1", "2" }, { "2", "4" }, { "3", "5" } };
+
+            Utils.MergeDictionaries(new Dictionary<string, string>[] { dict1, null });
+        }
+
+        [TestMethod]
+        public void MergeDictionaries_TwoDictionaries_NoOverlap()
+        {
+            var dict1 = new Dictionary<string, string> { { "1", "2" }, { "2", "4" }, { "3", "5" } };
```

Comment:
For readability, I'd suggest changing the value to something not a number.

You could do `{ 1, "one" }` for instance, and then later for the dupe tests use "one1" and "one2" as the values.

Reply:
done